terraform/modules/azure-nat-gateway/variables.tf

```
@@ -17,3 +17,9 @@ variable "subnet_names" {
   type        = list(string)
   description = "List of subnets (names) to associate with this gateway"
 }
+
+variable "outbound_ip_count" {
+  type        = number
+  description = "Amount of additional outbound Ips to use."
```

```suggestion
  description = "Number of additional outbound IPs to use."
```
Not sure about the "additional" since the default value is "1", thus:
```suggestion
  description = "Number of outbound IPs to use."
```

---

Good catch about the "additional": as I wrote in the PR body it is an implementation detail. As such it should not burden the end user.

So LGTM for your 2nd suggestion